vic/drivers/classic/src/read_initial_model_state.c

```diff
@@ -688,4 +692,90 @@ read_initial_model_state(FILE            *init_state,
             }
         }
     }
+
+    /******Check that soil moisture does not exceed maximum allowed************/
```

Comment:
Put which function in drivers/shared read_initial_model_state or vic_populate_model_state?  And do you mean shared_all or shared_image?

vic_populate_model_state() has some differences across drivers - cesm is different from image, let alone classic.  And read_initial_model_state() is not really shareable...

Or do you mean this section of code that I moved?